src/extension/utils/createAndOpenConfFile.ts

```diff
@@ -1,30 +1,28 @@
 import { workspace, Uri, window } from 'vscode'
 import { log, Sources } from '../utils/log'
-import { InputFormData } from '../types'
-
-type ConfFile = {
-  files?: string[]
-  verify?: [string]
-  solc?: string
-  link?: string[]
-  settings?: string[]
-  staging?: string
-  cache?: string
-  msg?: string
-} & Record<string, boolean | string>
+import { InputFormData, ConfFile } from '../types'
 
 function setAdditionalSetting(val?: string) {
   if (val === 'true' || !val) return true
   if (val === 'false') return false
-
+  if (/^[0-9]+$/.exec(val)) return Number(val)
+  const mapRegex = /^{(".+":".+")(,".+":".+")*}/g
+  if (mapRegex.exec(val)) {
+    return JSON.parse(val)
+  }
+  // for --settings flags:
+  const settingsRegex = /(-(.+)=(.+))(,(-(.+)=(.+)))*/g
+  if (settingsRegex.exec(val)) {
+    const squareBracketsRegex = /(\[|\])+/g
+    return val.replace(squareBracketsRegex, '').split(',')
```

Comment:
What are you trying to do here?
The settings should look like that (in the conf file):
```
"settings": [
        "-flag",
        "-other_flag=value",...
]
```

Reply:
because the settings are inside an array, I want to make sure that if a user inserts the settings manually and uses an array, I delete the square brackets from the string before I split it. example for manual `--settings`:
![Screen Shot 2022-05-23 at 12 23 08](https://user-images.githubusercontent.com/101042618/169787877-dc7e062b-aaeb-4360-824e-b33b687a9550.png)
.